ui/app/hooks/useCancelTransaction.js

```diff
@@ -19,27 +25,60 @@ import { getConversionRate, getSelectedAccount } from '../selectors';
  * @return {[boolean, Function]}
  */
 export function useCancelTransaction(transactionGroup) {
-  const { primaryTransaction, initialTransaction } = transactionGroup;
+  const { primaryTransaction } = transactionGroup;
   const gasPrice = primaryTransaction.txParams?.gasPrice?.startsWith('-')
     ? '0x0'
     : primaryTransaction.txParams?.gasPrice;
-  const { id } = initialTransaction;
+  const transaction = primaryTransaction;
   const dispatch = useDispatch();
   const selectedAccount = useSelector(getSelectedAccount);
   const conversionRate = useSelector(getConversionRate);
+  const defaultNewGasPrice = addHexPrefix(
+    multiplyCurrencies(gasPrice, 1.2, {
+      toNumericBase: 'hex',
+      multiplicandBase: 16,
+      multiplierBase: 10,
+    }),
+  );
+
   const cancelTransaction = useCallback(
     (event) => {
       event.stopPropagation();
-
+      dispatch(setCustomGasLimit('0x5208'));
+      dispatch(setCustomGasPriceForRetry(defaultNewGasPrice));
+      const tx = {
+        ...transaction,
+        txParams: {
+          ...transaction.txParams,
+          gas: '0x5208',
```

Comment:
What was the reason for overriding the `gas` and `value` here?

Reply:
I moved the code from here and preserved the `gas`:

![image](https://user-images.githubusercontent.com/364566/110040599-ee32b900-7cf7-11eb-9d70-1992ac54ce13.png)

for `value`, this is passed in now to ensure that when cancelling and merging the params, that for a cancellation, the `value` is 0. If not, then it would put the value of the transaction you want to cancel in the gas modal.

Reply:
I don't think setting the value to zero is necessary here, since it's set to zero here as well: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/index.js#L405

Now that I'm looking at this.... it looks like the limit is set here too! Overridden to be '0x5208' each time. Huh. Yet we show it in the UI despite ignoring it. That's unfortunate. Maybe a problem to solve in another PR though :shrug:

Reply:
Actually this does seem problematic :thinking: If the user enters an alternate gas limit, they'll be shown a different fee than they end up paying.

This is less concerning if they _raise_ the gas limit, because... why would a user do that? And at least they'd pay _less_ than they confirmed in that case.

But if they lower the gas limit below 0x5208, we'd be claiming it costs _less_ than it does. That seems bad.

It looks like it'd be relatively straightforward to pass the gas limit through here too though? Just add another parameter to `createCancelTransaction`. At least it looks like it'd be that simple :thinking:

Reply:
@Gudahtt updated `createCancelTransaction` and the CancelTransaction modal to handle custom gas limit